File: app/src/main/java/se/curtrune/lucy/screens/dev/ItemsTest.java
package se.curtrune.lucy.screens.dev;

import static se.curtrune.lucy.util.Logger.log;

import android.content.Context;

import se.curtrune.lucy.classes.item.Item;
import se.curtrune.lucy.persist.ItemsWorker;

public class ItemsTest {

    public static void testDeleteTree(Context context){
        log("...testDeleteTree()");
    }
    public static void createTree(Context context){
        log("...createFamilyItems()");
        Item rootItem = new Item("rootItem");
        rootItem = ItemsWorker.insertChild(ItemsWorker.getDailyRoot(context), rootItem, context);
        Item child1 = new Item("child1");
        child1 = ItemsWorker.insertChild(rootItem, child1, context);
        Item child2 = new Item("child2");
        child2 = ItemsWorker.insertChild(rootItem, child2, context);
        Item grandchild1 = new Item("grandChild1");
        grandchild1 = ItemsWorker.insertChild(child1, grandchild1, context);
        Item grandChild2 = new Item("grandChild2");
        grandChild2 = ItemsWorker.insertChild(child1, grandChild2, context);
        log("rootItem id", rootItem.getID());
    }
    public static void deleteTree(Item parent, Context context){
        log("...deleteTree(Item)", parent.getHeading());
        ItemsWorker.deleteTree(parent, context);
    }
    public static void deleteTree(long parentID, Context context){
        log("ItemTest.deleteTree(long, Context)");
        Item item = ItemsWorker.selectItem(parentID, context);
        if( item == null){
            log(" no item with that id found ");
            return;
        }
        ItemsWorker.VERBOSE = true;
        ItemsWorker.deleteTree(item, context);
    }
}

File: app/src/main/java/se/curtrune/lucy/screens/dev/ItemsTest.kt
package se.curtrune.lucy.screens.dev

import android.content.Context
import se.curtrune.lucy.app.LucindaApplication
import se.curtrune.lucy.classes.item.Item
import se.curtrune.lucy.util.Logger

object ItemsTest {
    private val repository = LucindaApplication.appModule.repository
    fun testDeleteTree(context: Context?) {
        Logger.log("...testDeleteTree()")
    }

    fun createTree(context: Context?) {
        Logger.log("...createFamilyItems()")
        var rootItem = Item("rootItem")
        val todoRoot = repository.getTodoRoot()
        rootItem = todoRoot?.let { repository.insertChild(it, rootItem) }!!
        var child1 = Item("child1")
        child1 = repository.insertChild(rootItem, child1)
        var child2  = Item("child2")
        child2 = repository.insertChild(rootItem, child2)
        var grandchild1 = Item("grandChild1")
        grandchild1 = repository.insertChild(child1, grandchild1)
        var grandChild2 = Item("grandChild2")
        grandChild2 = repository.insertChild(child1, grandChild2)
        println("rootItem id: ${rootItem.id}")
    }

    fun deleteTree(parent: Item, context: Context?) {
        Logger.log("...deleteTree(Item)", parent.heading)
        repository.deleteTree(parent)
    }

    fun deleteTree(parentID: Long, context: Context?) {
        Logger.log("ItemTest.deleteTree(long, Context)")
        val item = repository.selectItem(parentID)
        if (item == null) {
            Logger.log(" no item with that id found ")
            return
        }
        //ItemsWorker.VERBOSE = true
        repository.deleteTree(item)
    }
}
